SopraGameController.cpp

```diff
@@ -1,5 +1,114 @@
+        if(actionProbability < 0 || actionProbability > 1){
+            throw std::invalid_argument("Probability not between 0 an 1");
+        }
+
+        return rng(0.0, 1.0) < actionProbability;
+    }
+
+    auto getAllCrossedCells(const gameModel::Position &startPoint, const gameModel::Position &endPoint) ->
+    std::vector<gameModel::Position> {
+
+        std::vector<gameModel::Position> resultVect;
+
+        // check if cells are valid
+        if (gameModel::Environment::getCell(startPoint) == gameModel::Cell::OutOfBounds ||
+        gameModel::Environment::getCell(endPoint) == gameModel::Cell::OutOfBounds){
+            throw std::out_of_range("Source or destination of movement vector are out of bounds");
+        }
+
+        // add the start point to the result
+        resultVect.emplace_back(startPoint);
+
+        // check if start and end point are equal
+        if (startPoint == endPoint)
+            return resultVect;
+
+        // define an normalize the direction vector
+        gameModel::Vector dirVect(endPoint.x - startPoint.x, endPoint.y - startPoint.y);
+        dirVect.normalize();
+
+        // traverse the route between the two points
+        gameModel::Vector travVect(0, 0);
+        gameModel::Position lastCell = startPoint;
+        while ((travVect + startPoint) != endPoint) {
+
+            // found a new crossed cell
+            if ((travVect + startPoint) != lastCell) {
+                lastCell = travVect + startPoint;
+                resultVect.emplace_back(lastCell);
+            }
+
+            // make a step to travers the vector
+            travVect = travVect + (dirVect * 0.5);
+        }
+
+        // add the end point to the result
+        resultVect.emplace_back(endPoint);
+
+        return resultVect;
+    }
+
+    auto getDistance(const gameModel::Position &startPoint, const gameModel::Position &endPoint) -> int {
+
+        // check if cells are valid
+        if (gameModel::Environment::getCell(startPoint) == gameModel::Cell::OutOfBounds ||
+            gameModel::Environment::getCell(endPoint) == gameModel::Cell::OutOfBounds){
+
+            return -1;
+        }
+
+        // check if start and end point are equal
+        if (startPoint == endPoint){
+            return 0;
+        }
+
+        int totalDistance = 0;
+
+        // calc the differences within the components of the given points
+        int dX = abs(startPoint.x - endPoint.x);
+        int dY = abs(startPoint.y - endPoint.y);
+
+        // calculate the total distance
+        if (dX >= dY) {
+            totalDistance += dY;
+            totalDistance += dX-dY;
+        } else {
+            totalDistance += dX;
+            totalDistance += dY-dX;
+        }
+
+        return totalDistance;
+    }
+
+    auto getAllPossibleMoves(std::shared_ptr<gameModel::Player> actor, const gameModel::Environment &envi) -> std::vector<Move> {
+        return std::vector<Move>();
```

Comment:
```suggestion
        return {};
```
nur als Hinweis: so kurzer (darf gerne ignoriert werden da ja hier hoffentlich noch mehr Code kommt)